File: Practica 2/Parte 1/src/potencia.java
/**
 *
 * @author Rodrigo González Garcés 3BV2
 */
import java.util.Scanner;

public class potencia {
    public static void main(String[] args) {
        System.out.println("Programa que te indica la potencia más próxima a un número");
        Scanner scanner = new Scanner(System.in);
        System.out.println();

        System.out.print("Ingresa un número natural: ");
        int numero = Integer.parseInt(scanner.nextLine());
        System.out.println();

        int potencia = 1;
        int cuenta = 0;
        while (potencia * 2 < numero) {
            potencia *= 2;
            cuenta++;
        }
        System.out.println("La potencia mas cercana es 2^" + cuenta + " = " + potencia);
    }
}

File: Practica 3/parte1/potencia.py
print("Programa que calcula la potencia de 2 más próxima a un número")
numero = int(input("Ingresa un número natural: "))

potencia = 1
cuenta = 0

while potencia * 2 < numero:
    potencia *= 2
    cuenta += 1

print("La potencia más cercana es 2^" + str(cuenta) + " = " + str(potencia))
